app/models/client.rb

```diff
@@ -0,0 +1,130 @@
+  end
+
+  def find_all(entity, created_since = nil)
+    method = "get_#{entity.pluralize}"
+    created_since = (Date.today - Date.parse(created_since)).to_i
+    send(method, created_since)
+  end
+
+  def get_users(created_since = nil)
+    users = if created_since
+              @intercom_client.users.find_all(created_since: created_since)
+            else
+              @intercom_client.users.all
+            end
+
+    entities = users.collect do |user|
+      {
+        id: user.id,
+        user_id: user.user_id,
+        email: user.email,
+        name: user.name,
+        location_data: {
+          city_name: user.location_data.city_name,
+          country_name: user.location_data.country_name,
+          postal_code: user.location_data.postal_code,
+          region_name: user.location_data.region_name
+        },
+        company_id: user.companies[0].id,
+      }
+    end
+
+    entities
+  end
+
+  def get_companies(created_since = nil)
+    companies = if created_since
+                  @intercom_client.companies.find_all(created_since: created_since)
+                else
+                  @intercom_client.companies.all
+                end
+
+    entities = companies.collect do |company|
+      {
+        id: company.id,
+        name: company.name,
+        user_count: company.user_count,
+        monthly_spend: company.monthly_spend,
+      }
+    end
+
+    entities
+  end
+
+  def get_contacts(created_since = nil)
+    contacts = if created_since
+                  @intercom_client.contacts.find_all(created_since: created_since)
+                else
+                  @intercom_client.contacts.all
+                end
+
+    entities = contacts.collect do |contact|
+      {
+        id: contact.id,
+        user_id: contact.user_id,
+        email: contact.email,
+        name: contact.name,
+        location_data: {
+          city_name: contact.location_data.city_name,
+          country_name: contact.location_data.country_name,
+          postal_code: contact.location_data.postal_code,
+          region_name: contact.location_data.region_name
+        },
+      }
+    end
+
+    entities
+  end
+
+  def create(entity, params)
+    method = "create_#{entity.pluralize}"
+    send(method, params)
+  end
+
+  def create_users(params)
+    @intercom_client.users.create(params)
+  end
+
+  def create_companies(params)
+    @intercom_client.companies.create(params)
+  end
+
+  def create_contacts(params)
+    @intercom_client.contacts.create(params)
+  end
+
+  def update(entity, entity_id, params)
+    method = "update_#{entity.pluralize}"
+    send(method, entity_id, params)
+  end
+
+  def update_users(entity_id, params)
+    @intercom_client.users.create(params)
+  end
+
+  def update_companies(entity_id, params)
+    @intercom_client.companies.create(params)
+  end
+
+  def update_contacts(entity_id, params)
```

Comment:
When I tested contacts they would not have any `id`. According to API, we could find contacts by `id` or `user_id`. So if contact is not found by `id` then we would look up using `user_id`.